main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+)
+
+type Record struct {
+	ID        uint
+	FirstName string
+	LastName  string
+	PhoneN    string
+}
+
+func removeEntry() (errr string) {
+	var Records []Record
+	f, err := os.Open("Data.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	if err := json.NewDecoder(bufio.NewReader(f)).Decode(&Records); err != nil {
+		log.Fatal(err)
+	}
+	id := -1
+	fmt.Println("ID:")
+	fmt.Scanln(&id)
+	for i, v := range Records {
+		if v.ID == uint(id) {
+			Records = append(Records[:i], Records[i+1:]...)
+			f.Close()
+			f, _ = os.Create("Data.json")
+			defer f.Close()
```

Comment:
nit: The funny part here is that at this point of the code the file is still opened for reading. And on some operating systems (or special conditions) this may fail. This is why it's important to close files as early as possible. Or you can open them for read/write and close only once when all of the operations are done.